pkg/controller/master/backup/restore.go

```diff
@@ -631,6 +639,12 @@ func (h *RestoreHandler) createRestoredPVC(
 		Kind:     volumeSnapshotKindName,
 		Name:     dataSourceName,
 	}
+	/* Because we deprecate the storage class `longhorn` from v1.1.0,
+	 * we need to create the restore volume to the new default storage class
+	 */
+	if *spec.StorageClassName == "longhorn" && defaultSC != nil {
```

Comment:
I thought that we should block any volume created with storage `longhorn`.
The parameter `Migratable` is used for harvester volume (that does not create SM when access mode is RWX).
If we use the RWX volume that missing migratable, that would cause the VMs to do migration failure with this volume.

We should deny it from 1.1.0

Reply:
good catch, IIRC any modification to the longhorn SC will be reverted by the longhorn controller, and the only way to patch it is through the configmap.
So from the Harvester UI, we should either consider to hide the `longhorn` SC or make it not configurable, but users should still be able to use the existing longhorn SC if wanted and the hard coded change in here proabbly is not a good option,
If we choose the hide the original `longhorn` SC from the UI, the best way for VM restores to be migrated to the new SC is to allow select differnt SC during the restore(and the longhorn SC can't be selected).

Reply:
The old `longhorn` SC should not be used (the restore operation would create a new PVC for it) after we do not override (due to missing the `Migratable` parameter). There are some problems if we use the new volume and its PVC created by non-override `longhorn` SC.

Hardcoded `longhorn` here because we only need to prevent creating any volume with the old SC `longhorn` for the above reason. The other volumes with different SC could work well.

For now, I just assign the old `longhorn` to the new default SC. Other volumes keep restoring to the same SC. 
Agree that the best way to restore is that allows users to select what they want (sure, filter out something we do not want users to know).

Reply:
After testing, the old `longhorn` would modify the parameter `migratable` by dependency charts.
We do not need to change the restore behavior until starting to work on issue #2771.

Thanks for the opinions from @w13915984028, @guangbochen, @bk201